test/e2e/operator_test.go

```diff
@@ -757,6 +758,76 @@ func TestHostNetworkEndpointPublishingStrategy(t *testing.T) {
 	}
 }
 
+// TestHostNetworkPortBinding creates two ingresscontrollers on the same node
+// with different port bindings and verifies that both routers are available.
+func TestHostNetworkPortBinding(t *testing.T) {
+	// deploy first ingresscontroller with the default port bindings
+	name1 := types.NamespacedName{Namespace: operatorNamespace, Name: "host"}
+	ing1 := newHostNetworkController(name1, name1.Name+"."+dnsConfig.Spec.BaseDomain, nil, nil)
+	if err := kclient.Create(context.TODO(), ing1); err != nil {
+		t.Fatalf("failed to create the first ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ing1)
+
+	conditions := []operatorv1.OperatorCondition{
+		{Type: ingresscontroller.IngressControllerAdmittedConditionType, Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+		{Type: "DeploymentReplicasAllAvailable", Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+	}
+	err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name1, conditions...)
+	if err != nil {
+		t.Errorf("failed to observe expected conditions for the first ingresscontroller: %v", err)
+	}
+
+	// get first router's single replica
+	pods := &corev1.PodList{}
+	if err := kclient.List(context.TODO(), pods, client.InNamespace(operandNamespace)); err != nil {
+		t.Fatalf("failed to list the first ingresscontroller's PODs: %v", err)
+	}
+	var pod1 *corev1.Pod
+	for _, p := range pods.Items {
+		if strings.HasPrefix(p.Name, "router-"+name1.Name) {
+			pod1 = &p
+			break
+		}
+	}
+	if pod1 == nil {
+		t.Fatalf("failed to find the first ingresscontroller's POD: %v", err)
+	}
+
+	// create second ingresscontroller on the same node but with different port bindings
+	name2 := types.NamespacedName{Namespace: operatorNamespace, Name: "samehost"}
+	strategy := &operatorv1.HostNetworkStrategy{
+		HTTPPort:  9080,
+		HTTPSPort: 9443,
+		StatsPort: 9936,
+	}
+	// take the node placement of the first router
+	placement := &operatorv1.NodePlacement{
+		Tolerations: pod1.Spec.Tolerations,
+		NodeSelector: &metav1.LabelSelector{
+			MatchLabels: pod1.Spec.NodeSelector,
+		},
+	}
+	if placement.NodeSelector.MatchLabels == nil {
+		placement.NodeSelector.MatchLabels = map[string]string{}
+	}
+	placement.NodeSelector.MatchLabels["kubernetes.io/hostname"] = pod1.Spec.NodeName
```

Comment:
Why mutate `pod1`'s node selector?  Could this be simplified?
```suggestion
			MatchLabels: map[string]string{
				"kubernetes.io/hostname": pod1.Spec.NodeName,
			},
		},
	}
```